frontend/packages/yki/src/components/registration/examSession/PublicExamSessionListingFilters.tsx

```diff
@@ -18,28 +50,155 @@ export const PublicExamSessionFilters = () => {
     });
   };
 
-  // TODO Fixme
-  const searchButtonDisabled = false;
-  const handleEmptyBtnClick = scrollToSearch;
-  // eslint-disable-next-line no-console
-  const handleSearchBtnClick = () => console.log('empty btn clicked..');
+  const { filters, municipalities } = useAppSelector(examSessionsSelector);
+  const filteredExamSessions = useAppSelector(selectFilteredPublicExamSessions);
+  const {
+    language,
+    level,
+    municipality,
+    excludeFullSessions,
+    excludeNonOpenSessions,
+  } = filters;
+
+  const dispatch = useAppDispatch();
+  const onFilterChange = (filter: Partial<ExamSessionFilters>) => {
+    dispatch(setPublicExamSessionFilters(filter));
+  };
+
+  const handleEmptyBtnClick = () => {
+    dispatch(resetPublicExamSessionFilters());
+    onEmptyFilters();
+    scrollToSearch();
+  };
+
+  const languages = Object.values(ExamLanguage);
+  const translateLanguage = (language: string) =>
+    translateCommon('languages.' + language);
+
+  const levelToComboBoxOption = (v: ExamLevel) => ({
+    value: v.toString(),
+    label: translateCommon('levels.' + v.toString()),
+  });
+  const levelValues = Object.values(ExamLevel).map(levelToComboBoxOption);
+  const municipalityToComboBoxOption = (m: string) => ({
+    value: m,
+    label: m,
+  });
 
-  // TODO Renders just the headers for now, add actual filters + logic
   return (
```

Comment:
Täällä hieman toistoa filter-komponenttien toteutuksissa mutta ei ehkä oleellista tehdä sille mitään

Reply:
Mun mielestä tää on ihan siedettävällä tasolla. Ehkä vielä ihan selkeä näin, mutta toisaalta sitten jos halutaan lyödä lukkoon (myös muiden projektien osalta) tää rakenne että filtterin label on muotoa `<Typography variant="h3" component="label" ... />` ja sitten siihen kylkeen lyödään itse suodatuskomponentti, niin sitten voisi tarjoilla rakenteen shared-paketin alta.

Reply:
Toi vois olla ihan hyvä parannus kyllä, ei toki tän yhteydessä tarvii ruveta sellaista tekemään